### PR TITLE
PMM-3560 Add builds for pmm-admin and pmm-agent

### PR DESCRIPTION
**Pull request by EvgeniyPatlan**

### CODE REVIEW

File: build/bin/build-client-binary [hunk 1]
```diff
@@ -22,29 +22,30 @@ gobuild_component() {
...
+        build_command="go build -o \${binary_dir}/${result_file} ./src/${extract_path}/${component_path}"
+    fi
     docker run --rm -v ${tmp_dir}:/home/builder/tmp ${rpmbuild_docker_image} sh -c "
         set -o errexit
         set -o xtrace
         export GOOS=${GOOS:-linux}
         export GOARCH=${GOARCH:-amd64}
         export CGO_ENABLED=0
+	export PMM_RELEASE_PATH=\${binary_dir}/bin
+	export PMM_RELEASE_VERSION=${pmm_version}
```
Comment (AlekSi) [inline build/bin/build-client-binary]: Why we have `$` there, but `\$` on the previous line?

File: build/bin/build-client-source [hunk 2]
```diff
@@ -20,11 +20,45 @@ prepare_tarball() {
...
+            tar -zcvf ${repo_name}-${commit}.tar.gz ${repo_name}-${commit}
+            rm -rf ${repo_name}-${commit}
+            popd
+        else
+            git -C "${git_dir}" archive \
+                --format=tar.gz \
+                --prefix=${repo_name}-${commit}/ \
+                -o ${tarball} \
+                "${commit}"
+        fi
```
Comment (AlekSi) [inline build/bin/build-client-source]: My "bad smell" detector is on full alarm. What is going on there? Why we do another `git clone` if we already have everything cloned by submodules? Why we descent into submodules directories and remove stuff there? Really looks like an alien part there.

And Evgeniy – please add comments. The future generations (including you after two weeks of MySQL 8) will thank you.

File: build/bin/build-client-binary (hunk 1 above)
Comment (EvgeniyPatlan) [reply build/bin/build-client-binary]: it is needed to get correct path in docker:
with: /home/builder/
without: /home<local_user>

File: build/bin/build-client-source (hunk 2 above)
Comment (EvgeniyPatlan) [reply build/bin/build-client-source]: fixed.
But please note that once we don't do `git clone` and just work with data after submodule update we have the following issue during `make release` 
+ make release
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
env CGO_ENABLED=0 go build -v -o bin/pmm-agent -ldflags " \

File: build/bin/build-client-binary [hunk 3]
```diff
@@ -22,29 +22,34 @@ gobuild_component() {
...
     docker run --rm -v ${tmp_dir}:/home/builder/tmp ${rpmbuild_docker_image} sh -c "
         set -o errexit
         set -o xtrace
         export GOOS=${GOOS:-linux}
         export GOARCH=${GOARCH:-amd64}
         export CGO_ENABLED=0
+        export PMM_RELEASE_VERSION=${pmm_version}
+	export PMM_RELEASE_TIMESTAMP=$(date '+%s')
+	export PMM_RELEASE_FULLCOMMIT=$(git rev-parse HEAD)
+	export PMM_RELEASE_FULLCOMMIT=$(git describe --all --contains --dirty HEAD)
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply build/bin/build-client-binary]: And please add
```
export PMM_RELEASE_BRANCH=
```
or something like that to export variable with an empty value.

[the remark this answers is not in the document]
Comment (EvgeniyPatlan) [reply build/bin/build-client-binary]: done

**Comment by rnovikovP** [comment]
It's good to know we have this  .. but this is 100% out of my   knowledge so can't review

**Pull request merged**